### PR TITLE
scheduler: Cancel restarts when removing jobs.

### LINKED ISSUE #610: scheduler: Cancel job restart timers
When scaling a formation process type down, jobs which are scheduled to start in the future should have their timers cancelled.

### PR DESCRIPTION
**Pull request by archSeer**
Fixes #610.

Signed-off-by: Blaž Hrastnik <blaz.hrast@gmail.com>

### CODE REVIEW

File: controller/scheduler/main.go [hunk 1]
```diff
@@ -746,10 +757,19 @@ func (f *Formation) jobType(job *host.Job) string {
...
-func (s sortJobs) Len() int           { return len(s) }
-func (s sortJobs) Less(i, j int) bool { return s[i].startedAt.Sub(s[j].startedAt) > 0 }
-func (s sortJobs) Swap(i, j int)      { s[i], s[j] = s[j], s[i] }
-func (s sortJobs) Sort()              { sort.Sort(s) }
+func (s sortJobs) Len() int { return len(s) }
+func (s sortJobs) Less(i, j int) bool {
+	switch {
+	case s[i].timer != nil && s[j].timer == nil:
+		return true
+	case s[i].timer == nil && s[j].timer != nil:
```
Comment (titanous) [inline controller/scheduler/main.go]: Copying this comment from the other PR: Does this need a mutex? Where is sort called from?

Comment (lmars) [inline controller/scheduler/main.go]: `sortJobs` is used then removing jobs due to scaling, whereas `Job.timer` is potentially written to when the host job crashes (which will be in another goroutine), so I think it does need a lock here.

**Comment by titanous** [comment]
What do we need to get this merged?

**Comment by archSeer** [comment]
I think adding the locks was the last thing needed.

**Comment by titanous** [comment]
LGTM

**Pull request merged**